lib/index.js

```diff
@@ -440,11 +440,6 @@ class Stream extends Duplex {
     if (!this.source.writable) {
       cb(new Error('socket not writable'))
     }
-
-    this.source.write(data, encoding, () => {
```

Comment:
You definitely want to leave `this.source.write` on and just add `cb` instead of a lambda here.

Reply:
Got it. I have committed the code again. 
Thanks.